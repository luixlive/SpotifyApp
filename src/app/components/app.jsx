import PropTypes from 'prop-types';
import React from 'react';

import Header from './header';
import Footer from './footer';
import { SizeDetector } from './hoc';

export const App = props => (
  <div>
    <Header isUserAuthenticated={false} />
    {props.children}
    <Footer />
  </div>
);

App.propTypes = {
  children: PropTypes.arrayOf(PropTypes.element).isRequired,
};

export default SizeDetector(App);
