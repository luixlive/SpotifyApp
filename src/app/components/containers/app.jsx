import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import React, { Component } from 'react';
import { withRouter } from 'react-router-dom';

import { Footer } from './../';
import { Header } from './../containers';
import { loadUser } from './../../actions/user';
import ScreenLoader from './../screen_loader';
import { SizeDetector } from './../hoc';

export class PureApp extends Component {
  componentDidMount() {
    this.props.loadUser();
  }

  renderBody() {
    if (this.props.userLoaded) {
      return this.props.children;
    }
    return <ScreenLoader />;
  }

  render() {
    return (
      <div>
        <Header />
        {this.renderBody()}
        <Footer />
      </div>
    );
  }
}

PureApp.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
  ]).isRequired,
  loadUser: PropTypes.func.isRequired,
  userLoaded: PropTypes.bool.isRequired,
};

const mapStateToProps = ({ user }) => ({ userLoaded: user.userLoaded });

const mapDispatchToProps = dispatch => ({
  loadUser: () => dispatch(loadUser()),
});

export const ConnectedApp =
  connect(mapStateToProps, mapDispatchToProps)(PureApp);

export default withRouter(SizeDetector(ConnectedApp));
