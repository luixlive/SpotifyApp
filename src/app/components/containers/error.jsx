import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import React, { Component } from 'react';

import clearError from './../../actions/error';
import { loadUser } from './../../actions/user';
import { Message } from './../templates';

export class PureError extends Component {
  loadUserAndClearError = () => {
    this.props.clearError();
    this.props.loadUser();
  }

  render() {
    return (
      <Message title="Error :(">
        {this.props.error} Go back to
        <Link onClick={this.loadUserAndClearError} to="/"> Home</Link>.
      </Message>
    );
  }
}

PureError.propTypes = {
  clearError: PropTypes.func.isRequired,
  error: PropTypes.string.isRequired,
  loadUser: PropTypes.func.isRequired,
};

const mapStateToProps = ({ error }) => ({ error });

const mapDispatchToProps = dispatch => ({
  clearError: () => dispatch(clearError()),
  loadUser: () => dispatch(loadUser()),
});

export default connect(mapStateToProps, mapDispatchToProps)(PureError);
