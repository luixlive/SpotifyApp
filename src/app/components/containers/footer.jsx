import React from 'react';
import { Segment } from 'semantic-ui-react';

const Footer = () => (
  <Segment className="footer" textAlign="center">
    <p>
      Favicon made by
      <a
        href="https://www.flaticon.com/authors/gregor-cresnar"
        rel="noopener noreferrer"
        target="_blank"
      >
        {' Gregor Cresnar '}
      </a>
      from
      <a
        href="https://www.flaticon.com"
        rel="noopener noreferrer"
        target="_blank"
      >
        {' www.flaticon.com'}
      </a>.
    </p>
  </Segment>
);

export default Footer;
