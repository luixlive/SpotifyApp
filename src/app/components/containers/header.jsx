import { Button, Grid, Header, Segment } from 'semantic-ui-react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import React, { Component } from 'react';

import { logoutUser } from './../../actions/user';

export class PureHeader extends Component {
  getLoginButtonText = () => {
    if (this.props.userAuthenticated) {
      return 'Logout';
    }
    return this.props.deviceMobile ? 'Log in' : 'Log in with Spotify';
  };

  render() {
    return (
      <Segment color="green" textAlign="center">
        <Grid columns={this.props.deviceMobile ? 2 : 3}>
          {!this.props.deviceMobile && <Grid.Column width={4} />}
          <Grid.Column verticalAlign="middle" width={8}>
            <Header as={this.props.deviceMobile ? 'h2' : 'h1'} >
              Music Stats
            </Header>
          </Grid.Column>
          <Grid.Column width={this.props.deviceMobile ? 8 : 4}>
            <Button
              inverted
              color="green"
              floated="right"
              onClick={() => {
                if (this.props.userAuthenticated) {
                  this.props.logoutUser();
                } else {
                  window.open('/api/authentication/spotify', '_self');
                }
              }}
            >
              {this.getLoginButtonText()}
            </Button>
          </Grid.Column>
        </Grid>
      </Segment>
    );
  }
}

PureHeader.propTypes = {
  deviceMobile: PropTypes.bool.isRequired,
  userAuthenticated: PropTypes.bool.isRequired,
  logoutUser: PropTypes.func.isRequired,
};

const mapStateToProps = ({ deviceMobile, user }) => ({
  deviceMobile,
  userAuthenticated: user.userAuthenticated,
});

const mapDispatchToProps = dispatch => ({
  logoutUser: () => dispatch(logoutUser()),
});

export default connect(mapStateToProps, mapDispatchToProps)(PureHeader);
