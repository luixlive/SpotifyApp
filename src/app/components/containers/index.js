import App from './app';
import Error from './error';
import Footer from './footer';
import Header from './header';
import Login from './login';
import UserStats from './user_stats';

export { App, Error, Footer, Header, Login, UserStats };
