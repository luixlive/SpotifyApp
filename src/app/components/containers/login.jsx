import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import React from 'react';
import { Redirect } from 'react-router-dom';

import Message from './../templates';

export const PureLogin = (props) => {
  if (props.userAuthenticated) {
    return <Redirect to="/" />;
  }
  return (
    <Message title="Welcome">
      Please log in with Spotify to start.
    </Message>
  );
};

PureLogin.propTypes = { userAuthenticated: PropTypes.bool.isRequired };

const mapStateToProps = ({ user }) => ({
  userAuthenticated: user.userAuthenticated,
});

const mapDispatchToProps = () => ({});

export default connect(mapStateToProps, mapDispatchToProps)(PureLogin);
