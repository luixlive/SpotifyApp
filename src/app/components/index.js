import AppConnected from './app';
import AppFooterConnected from './footer';
import AppHeaderConnected from './header';
import LoginConnected from './login';
import Router from './router';
import UserStatsConnected from './user_stats';

export {
  AppConnected,
  AppFooterConnected,
  AppHeaderConnected,
  LoginConnected,
  Router,
  UserStatsConnected,
};
