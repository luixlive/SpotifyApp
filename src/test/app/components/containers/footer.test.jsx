import React from 'react';
import renderer from 'react-test-renderer';

import { Footer } from './../../../../app/components/containers';

describe('App Components - Footer', () => {
  it('renders', () => {
    const rendered = renderer.create(<Footer />).toJSON();
    expect(rendered).toMatchSnapshot();
  });
});
