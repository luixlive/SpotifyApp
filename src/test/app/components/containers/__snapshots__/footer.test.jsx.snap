// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Components - Footer renders 1`] = `
<div
  className="ui center aligned segment footer"
>
  <p>
    This app uses the
    <a
      href="https://www.spotify.com"
      rel="noopener noreferrer"
      target="_blank"
    >
       Spotify 
    </a>
    API. Favicon made by
    <a
      href="https://www.flaticon.com/authors/gregor-cresnar"
      rel="noopener noreferrer"
      target="_blank"
    >
       Gregor Cresnar 
    </a>
    from
    <a
      href="https://www.flaticon.com"
      rel="noopener noreferrer"
      target="_blank"
    >
       flaticon
    </a>
    . Contact me on
    <a
      href="https://twitter.com/LuisAlfonsoCHA"
      rel="noopener noreferrer"
      target="_blank"
    >
       twitter
    </a>
    .
  </p>
</div>
`;
